docs/reference/buildx.md

```diff
@@ -0,0 +1,32 @@
+# buildx
```

Comment:
Note; moving the docs to be in `docs/reference`; this is more in line with other repositories that generate yaml docs (for integration in docs.docker.com); having these in a separate directory keeps the top `docs/` directory available for other (non-reference) docs. If we don't anticipate other docs to be added, we could move to the top-level `docs/`